### PR TITLE
Don't fire 'error' for already-aborted transition

### PR DESCRIPTION
**Pull request by ef4**
This closes https://github.com/emberjs/ember.js/issues/4573

### CODE REVIEW

**Comment by stefanpenner** [comment]
@ef4 this code change may be a hazard, as assertions which fire during `this.transitionTo` are now dropped, as the transition is marked as aborted.

**Comment by rwjblue** [comment]
@stefanpenner - Please open an issue with a repro so we have the ability to address, I do not fully understand the scenario you are describing...

File: test/tests/router_test.js [hunk 1]
```diff
@@ -1564,6 +1564,44 @@ test("Errors shouldn't be handled after proceeding to next child route", functio
...
+    match("/slow_failure").to('slow_failure');
+    match("/good").to('good');
+  });
+
+  handlers = {
+    slow_failure: {
+      model: function() {
+	return new Promise(function(res, rej){
+	  router.transitionTo('good');
+	  rej();
```
[the remark this answers is not in the document]
Comment (rwjblue) [reply test/tests/router_test.js]: I dunno, maybe I'm missing something obvious but I don't see a syntax error here....

[the remark this answers is not in the document]
Comment (rwjblue) [reply test/tests/router_test.js]: Either way though, opening an issue with a repro is the best (possibly only?) way to ensure we actually fix it...

**Pull request merged**